lib/osml/model_runner/roles/mr_task_role.ts

```diff
@@ -74,72 +85,206 @@ export class MRTaskRole extends Construct {
+        `arn:${this.partition}:sqs:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SQS_REGION_STATUS_QUEUE}DLQ`
+      ]
+    });
+
+    // Add permissions for S3 permissions
+    const s3PolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: [
+        "s3:GetBucketAcl",
+        "s3:ListBucket",
+        "s3:GetBucketLocation",
+        "s3:GetObject",
+        "s3:GetObjectAcl",
+        "s3:PutObject",
+        "s3:DeleteObject"
+      ],
+      resources: [`arn:${this.partition}:s3:::*`]
+    });
+
+    // Add permissions for SNS permissions
+    const snsPolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: ["sns:Publish"],
+      resources: [
+        `arn:${this.partition}:sns:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SNS_IMAGE_STATUS_TOPIC}`,
+        `arn:${this.partition}:sns:${props.account.region}:${props.account.id}:${this.mrDataplaneConfig.SNS_REGION_STATUS_TOPIC}`
+      ]
+    });
+
+    // Add permissions for dynamodb permissions
+    const ddbPolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: [
+        "dynamodb:BatchGetItem",
+        "dynamodb:BatchWriteItem",
+        "dynamodb:PutItem",
+        "dynamodb:ListTables",
+        "dynamodb:DeleteItem",
+        "dynamodb:GetItem",
+        "dynamodb:Query",
+        "dynamodb:Scan",
+        "dynamodb:UpdateItem",
+        "dynamodb:UpdateTable"
+      ],
+      resources: [
+        `arn:${this.partition}:dynamodb:${props.account.region}:${props.account.id}:table/${this.mrDataplaneConfig.DDB_JOB_STATUS_TABLE}`,
+        `arn:${this.partition}:dynamodb:${props.account.region}:${props.account.id}:table/${this.mrDataplaneConfig.DDB_FEATURES_TABLE}`,
+        `arn:${this.partition}:dynamodb:${props.account.region}:${props.account.id}:table/${this.mrDataplaneConfig.DDB_ENDPOINT_PROCESSING_TABLE}`,
+        `arn:${this.partition}:dynamodb:${props.account.region}:${props.account.id}:table/${this.mrDataplaneConfig.DDB_REGION_REQUEST_TABLE}`
+      ]
+    });
+
+    // Add permission for autoscaling ECS permissions
+    const autoScalingEcsPolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: ["ecs:DescribeServices", "ecs:UpdateService"],
+      resources: [
+        `arn:${this.partition}:ecs:${props.account.region}:${props.account.id}:cluster/${this.mrDataplaneConfig.MR_CLUSTER_NAME}`,
+        `arn:${this.partition}:ecs:${props.account.region}:${props.account.id}:service/${this.mrDataplaneConfig.MR_CLUSTER_NAME}/*`
+      ]
+    });
+
+    // Add permission for autoscaling CW permissions
+    const autoScalingCwPolicyStatement = new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: ["cloudwatch:DescribeAlarms"],
+      resources: [`*`]
+    });
+
+    // // Add permissions for cloudwatch permissions
```

Comment:
Do we want to remove this commented out section?

Reply:
Good point, removing it.